torch/autograd/gradcheck.py

```diff
@@ -17,12 +18,12 @@ def zero_gradients(x):
 
 
 def make_jacobian(input, num_out):
-    if isinstance(input, torch.Tensor):
+    if is_tensor_like(input):
         if not input.is_floating_point() and not input.is_complex():
             return None
         if not input.requires_grad:
             return None
-        return torch.zeros(input.nelement(), num_out, dtype=input.dtype)
+        return input.new_zeros((input.nelement(), num_out), dtype=input.dtype, layout=torch.strided)
```

Comment:
Huh, weird that you need an explicit `layout` here.

Reply:
If the input is sparse or MKLDNN (for example), `torch.zeros` produces a strided array.